ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+
+                        if (finalFileExists)
+                        {
+                            Log.Error($"BenzingaDataConverter.WriteToFile(): Moving backup file: {existingReferenceFileBackupPath} back to original location: {referenceFilePath}");
+                            File.Move(existingReferenceFileBackupPath, referenceFilePath);
+                        }
+
+                        continue;
+                    }
+
+                    // Clean up after ourselves so that we don't accidentally run out of storage space
+                    File.Delete(existingReferenceFileBackupPath);
+
+                    createdReference = true;
+                }
+
+                if (!createdReference)
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Skipping news article {article.Id} because no tickers are associated with it");
+                    continue;
+                }
+
+                var finalArticleFile = Path.Combine(finalDirectory.FullName, $"{article.Id}.json");
+                var tempArticleFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.json");
+                var backupArticleFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.json");
+                var articleExists = File.Exists(finalArticleFile);
+
+                Log.Trace($"BenzingaDataConverter.WriteToFile(): Writing article contents to temp: {tempArticleFile}");
+                File.WriteAllText(tempArticleFile, JsonConvert.SerializeObject(article, Newtonsoft.Json.Formatting.None));
+
+                if (articleExists)
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Article already exists. Moving from: {finalArticleFile} - to backup: {backupArticleFile}");
+                    File.Move(finalArticleFile, backupArticleFile);
+                }
+
+                try
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Moving from temp to: {finalArticleFile}");
+                    File.Move(tempArticleFile, finalArticleFile);
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, $"BenzingaDataConverter.WriteToFile(): Failed to move file to: {finalArticleFile}");
+
+                    if (articleExists)
+                    {
+                        Log.Error($"BenzingaDataConverter.WriteToFile(): Restoring backup article file: {backupArticleFile} to original location: {finalArticleFile}");
+                        File.Move(backupArticleFile, finalArticleFile);
+                    }
+                }
+            }
+
+            // Now compress all of the data we have for a given day
+            var compressedFinal = new FileInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyyMMdd}.zip"));
+            var compressedFinalBackup = new FileInfo(Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.zip"));
+
+            if (compressedFinal.Exists)
```

Comment:
not applicable here, but an interesting and potentially headache-inducing side effect of using `FileInfo` is that they only load their data once -- the first time you access anything derived from the file's headers/options. You can update an instance of `FileInfo` manually by calling the `Refresh()` function. Here's an [SO thread](https://github.com/dotnet/corefx/issues/42447) discussing it along with the [method documentation](https://docs.microsoft.com/en-us/dotnet/api/system.io.filesysteminfo.refresh?view=netframework-4.8). You'll note this is pointing at `FileSystemInfo` -- which is the base class of `FileInfo` and he's the one that defines the `Refresh` function.

EDIT: You're totally all over this -- didn't see a usage of `FileSystemInfo.Refresh()` until hitting enter on this one. Leaving here for posterity.